packages/jupyter-ai/src/components/pending-messages.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect } from 'react';
+
+import { Box } from '@mui/material';
+import { AiService } from '../handler';
+import { ChatMessageHeader } from './chat-messages';
+
+type PendingMessagesProps = {
+  messages: AiService.PendingMessage[];
+};
+
+type PendingMessageElementProps = {
+  text: string;
+  ellipsis: boolean;
+};
+
+function PendingMessageElement(props: PendingMessageElementProps): JSX.Element {
+  const [dots, setDots] = useState('');
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setDots(dots => (dots.length < 3 ? dots + '.' : ''));
+    }, 500);
+
+    return () => clearInterval(interval);
+  }, []);
+
+  let text = props.text;
+  if (props.ellipsis) {
+    text = props.text + dots;
+  }
+
+  return (
+    <div>
+      {text.split('\n').map((line, index) => (
+        <p key={index} style={{ lineHeight: 0.6 }}>
+          {line}
+        </p>
+      ))}
+    </div>
+  );
+}
+
+export function PendingMessages(
+  props: PendingMessagesProps
+): JSX.Element | null {
+  if (props.messages.length === 0) {
+    return null;
+  }
```

Comment:
This early return causes the `useState` and `useEffect` hooks below to be called conditionally. To address this:

1. Define `agentMessage` as a state variable
2. Add this effect hook:

```ts
useEffect(() => {
  if (props.messages.length === 0) {
    setAgentMessage(null)
  else {
    const lastMessage = props.messages[props.messages.length - 1]
    ... // set agent message to a dummy object with lastMessage
  }
}, [props.messages])
```

3. Remove the highlighted block and add this right before the `return` statement on line 73:

```ts
if (!agentMessage) {
  return null
}

...
```